ecs-agent/doctor/doctor_test.go

```diff
@@ -0,0 +1,195 @@
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	TEST_CLUSTER      = "test-cluster"
+	TEST_INSTANCE_ARN = "test-instance-arn"
+)
+
+type trueHealthcheck struct{}
+
+func (tc *trueHealthcheck) RunCheck() HealthcheckStatus                   { return HealthcheckStatusOk }
+func (tc *trueHealthcheck) SetHealthcheckStatus(status HealthcheckStatus) {}
+func (tc *trueHealthcheck) GetHealthcheckType() string                    { return HealthcheckTypeAgent }
+func (tc *trueHealthcheck) GetHealthcheckStatus() HealthcheckStatus {
+	return HealthcheckStatusInitializing
+}
+func (tc *trueHealthcheck) GetLastHealthcheckStatus() HealthcheckStatus {
+	return HealthcheckStatusInitializing
+}
+func (tc *trueHealthcheck) GetHealthcheckTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+func (tc *trueHealthcheck) GetStatusChangeTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+func (tc *trueHealthcheck) GetLastHealthcheckTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+
+type falseHealthcheck struct{}
+
+func (fc *falseHealthcheck) RunCheck() HealthcheckStatus                   { return HealthcheckStatusImpaired }
+func (fc *falseHealthcheck) SetHealthcheckStatus(status HealthcheckStatus) {}
+func (fc *falseHealthcheck) GetHealthcheckType() string                    { return HealthcheckTypeAgent }
+func (fc *falseHealthcheck) GetHealthcheckStatus() HealthcheckStatus {
+	return HealthcheckStatusInitializing
+}
+func (fc *falseHealthcheck) GetLastHealthcheckStatus() HealthcheckStatus {
+	return HealthcheckStatusInitializing
+}
+func (fc *falseHealthcheck) GetHealthcheckTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+func (fc *falseHealthcheck) GetStatusChangeTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+func (fc *falseHealthcheck) GetLastHealthcheckTime() time.Time {
+	return time.Date(1974, time.May, 19, 1, 2, 3, 4, time.UTC)
+}
+
+func TestNewDoctor(t *testing.T) {
+	trueCheck := &trueHealthcheck{}
+	falseCheck := &falseHealthcheck{}
+	healthchecks := []Healthcheck{trueCheck, falseCheck}
+	newDoctor, _ := NewDoctor(healthchecks, TEST_CLUSTER, TEST_INSTANCE_ARN)
+	assert.Len(t, newDoctor.healthchecks, 2)
+}
+
+func TestAddHealthcheck(t *testing.T) {
+	newDoctor := Doctor{}
+	assert.Len(t, newDoctor.healthchecks, 0)
+	newTestHealthcheck := &trueHealthcheck{}
+	newDoctor.AddHealthcheck(newTestHealthcheck)
+	assert.Len(t, newDoctor.healthchecks, 1)
+}
+
+func TestGetCluster(t *testing.T) {
+	clusterName := "test-cluster"
+	newDoctor := Doctor{cluster: clusterName}
+	assert.Equal(t, newDoctor.GetCluster(), clusterName)
```

Comment:
nit: `assert.Equal` function takes the expected value before the actual value. 

[func Equal(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool](https://pkg.go.dev/github.com/stretchr/testify@v1.8.2/assert#Equal)

Not too big a deal but if the test fails then the values against "Expected" and "Actual" will be printed incorrectly.